### ISSUE TITLE
Add documentation about using gc to src/plugins/README.md

### ISSUE THREAD

**Issue by henrikingo** [impress/impress.js#677]
Since all plugins must support teardown, they must use `src/lib/gc.js` for modifying DOM, including adding event listeners. This needs to be documented in the plugins readme.

(Until that happens, every existing plugin uses the gc library, so just looking at existing code should help, even if it may not be obvious *why* they do that.)

**Comment by simplePCBuilding** [comment]
@henrikingo I will write a new plugin documentation for V3 update. I am working on a completely new plugins api. I know this breaks all plugins, but I think making the usage and creation of plugins easier should be a good thing. I will be adding a migration guide as well

**Comment by henrikingo** [comment]
Yes so nowadays there are official or well known plugin mechanisms also for browser side. Can we use one of those? That would be very much in the spirit of impressjs to use modern and standard JavaScript.

**Comment by simplePCBuilding** [comment]
@henrikingo Which ones do you mean, I don't remember hearing of any...
